### ISSUE TITLE
An error occurred:   --> Value was either too large or too small for an Int32.

### ISSUE THREAD

**Issue by ArnoutVerbeken** [pgonzal/yamster#1]
Using Yamster for a long time now without any real issues.
Since a couple of days, I can't sync anymore.  I receive the following error:
An error occurred:
  --> Value was either too large or too small for an Int32.

I don't know how to troubleshoot.  I can't find any logfiles.

Please let me know how I can help debug.

**Comment by octogonz** [comment]
A user! :D I didn't know people still used this app.

I'll see if I can repro it. It may be specific to your company's data set.

**Comment by octogonz** [comment]
Yeah, Yamster still works fine for me. Try this:

1. Make sure you've closed Yamster.exe and any other Yamster-related stuff
2. Go to your folder where Yamster is installed (e.g. `C:\Program Files (x86)\Yamster`)
3. Temporarily rename `Yamster.Core.dll` --> `Old-Yamster.Core.dll`
4. Unzip the two files from [YamsterDebug.zip](https://github.com/octogonz/yamster/files/2852805/YamsterDebug.zip) into that folder (`Yamster.Core.dll` and `Yamster.Core.dll.pdb`).  This will enable debugging traces.
5. Run the application and reproduce your error.  Reply to this GitHub issue with the new error text.

Hopefully this time the message will include a call stack that should help us to figure out where is the `Int32` that's causing the trouble.

**Comment by ArnoutVerbeken** [comment]
Here is the output with debug on:
An error occurred:
  --> Value was either too large or too small for an Int32.

   at Yamster.Core.MessagePuller.Process() in D:\GitRepos\yamster\Source\Yamster.Core\Sync\MessagePuller.cs:line 183
   at Yamster.SyncWindow.SyncOnce()
   at Yamster.SyncWindow.OnPollTimer()

**Comment by octogonz** [comment]
Hmmm... this is a summary exception.  Could you try one more time with this one?

[YamsterDebug2.zip](https://github.com/octogonz/yamster/files/2857306/YamsterDebug2.zip)

I suspect that this is going to end up being something in the protocol, and we might need to use [Fiddler](https://www.telerik.com/download/fiddler) to collect the server response. But maybe we'll get lucky with the stack trace. :-)

(BTW you can ping me on [Gitter](https://gitter.im/) if you want to diagnose this more quickly.)

**Comment by ArnoutVerbeken** [comment]
Output with debug2:
```
An error occurred:
  --> Value was either too large or too small for an Int32.

  --> Message: Value was either too large or too small for an Int32.
Stack trace:    at System.Convert.ToInt32(Int64 value)
   at System.Int64.System.IConvertible.ToInt32(IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType, IFormatProvider provider)
   at System.Convert.ChangeType(Object value, Type conversionType)
   at Yamster.Core.SQLite.SQLiteMapper.<>c__DisplayClass21_0`1.<QueryScalar>b__0() in D:\GitRepos\yamster\Source\Yamster.Core\SQLite\SQLiteMapper.cs:line 386
   at Yamster.Core.SQLite.SQLiteMapper.WrapDatabaseExceptions[T](Func`1 action) in D:\GitRepos\yamster\Source\Yamster.Core\SQLite\SQLiteMapper.cs:line 98
   at Yamster.Core.SQLite.SQLiteMapper.QueryScalar[T](String sqlQuery) in D:\GitRepos\yamster\Source\Yamster.Core\SQLite\SQLiteMapper.cs:line 372
   at Yamster.Core.MessagePuller.<ProcessSpanAsync>d__45.MoveNext() in D:\GitRepos\yamster\Source\Yamster.Core\Sync\MessagePuller.cs:line 661
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.GetResult()
   at Yamster.Core.MessagePuller.<ProcessAsync>d__43.MoveNext() in D:\GitRepos\yamster\Source\Yamster.Core\Sync\MessagePuller.cs:line 381

   at Yamster.Core.MessagePuller.Process() in D:\GitRepos\yamster\Source\Yamster.Core\Sync\MessagePuller.cs:line 184
   at Yamster.SyncWindow.SyncOnce()
   at Yamster.SyncWindow.OnPollTimer()
```

I pinged you on Gitter. ;)

**Comment by octogonz** [comment]
That's super helpful!  I think I found the problem.  Your company has been using Yammer so much that your message IDs have somehow exceeded the maximum `Int32` size of 2,147,483,647. :-P Yamster is supposed to use `Int64` for these IDs, but there were a few places that I had missed.

Please try this fix and confirm whether it solves your issue: [YamsterDebug3.zip](https://github.com/octogonz/yamster/files/2861545/YamsterDebug3.zip)

If it works, I can publish it as a proper release.  Thanks!

**Comment by ArnoutVerbeken** [comment]
I confirm this is working now!  My yams are flowing in again.  Thank you very much for the quick fix!

**Comment by octogonz** [comment]
Thanks again for reporting this. I've published a new release [Yamster 2.9.3](https://github.com/octogonz/yamster/releases/tag/2.9.3) with the fix.
